tests/language-feature/bitfield/generic.slang

```diff
@@ -0,0 +1,29 @@
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-cpu -output-using-type
+
+// CHECK:      123
+// CHECK-NEXT: 4567
+// CHECK-NEXT: 0
+// CHECK-NEXT: 0
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<uint> outputBuffer;
+
+struct S<T> {
+    int foo : 8;
+    uint bar : 24;
+    T baz;
+};
```

Comment:
I think we should be sensitive about which buts will be used first. The first variable could use the most significant its or least significant bits. Because this behavior is not standardized in C/C++, gcc and vsc provides an option to change it. By default gcc and vsc have different behavior. This issue had been a big pain during the playstation 3 generation because Sony was using gcc while xbox was using visual studio.

Reply:
I wonder if even this could be done as an annotation on the fields themselves..

Reply:
https://github.com/shader-slang/slang/issues/3646